asset-transfer-private-data/chaincode-go/private_asset_queries.go

```diff
@@ -47,8 +47,9 @@ func (s *SmartContract) ReadAssetPrivateDetails(ctx contractapi.TransactionConte
 		return nil, fmt.Errorf("failed to read from asset details %v", err)
 	}
 	if assetDetailsJSON == nil {
-		log.Printf("AssetPrivateDetails for %v does not exist in collection %v", assetID, collection)
-		return nil, nil
+		message := fmt.Sprintf("AssetPrivateDetails for %s does not exist in collection %s", assetID, collection)
+		log.Print(message)
+		return nil, fmt.Errorf(message)
```

Comment:
In the last PR, while developing JS app, I switched to the pattern of reads to return empty data when Data_Not_Found: so, all the ReadXXX functions  return empty data , instead of throwing Error. As a client, I believe that was more intuitive.

IIRC when the read call access a collection that it does not have access to, the error is returned to client, to indicate Access error.

if we really need this pattern to throw Err, then the change should be for all ReadXX and the unit test in #290 need rework.

Reply:
I do prefer to return an error if the asset is not read (all the other samples do this). If nothing else, showing how to handle an expected error in the application, will help users to start thinking about how to handle all errors.

But I agree that the change is out of scope of this PR. I've removed it for now, I'll defer to Sijo since he has been doing the majority of the work for this sample.

Reply:
Sounds good. There is not an 'update' per se in this sample, but you could test failure of any of the conditions validated against in verifyAgreement() or verifyClientOrgMatchesPeerOrg().